documentation/specs/rar-as-service.md

```diff
@@ -0,0 +1,216 @@
+
+### Execute RAR task
+
+Execution of RAR task should be done in same way as it is done now.
+
+There is already some layer of separation between Task interface and actual execution method. We will leverage this, and put the decision logic if to run locally or not into the "wrapper" method and so we will not have to modify this and in server-side execution we will directly call the internal execution method.
+
+There is one big concern and that is how to handle multiple requests at once. As right now, RAR task is not prepared for multi-thread use.
+
+### Shutdown RAR node
+
+If the user wants the node to not be reused, we have the ensure that node will be killed after the build ends. This should be done after the compilation end by main MSBuild node.
+
+The RAR node, also has to support accepting of already established commands for MSBuild nodes (for example Shutdown command). This will be done by creating two pipes inside node, one will be for communication about RAR commands and second one for the servicing communication. 
+
+### Execute task in MSBuild node
+
+User opted out of using the RAR nodes so we will execute the RAR task in the MSBuild node (as it is right now).
+
+### Other
+
+The new RAR node will not count toward total maximum CPU count specified by _/maxCpuCount_ switch, since the RAR task is taxing on IO operations not so much on CPU time. If we took one node from each instance of MSBuild it would lead to drastic decrease in performance.
+
+The RAR task will be affected by the _/m_ switch. When we run in single node mode, it will implicitly say that we want to run the task inside current process. User would have to explicitly say that they want to use the RAR node.
+
+__NOTE:__ The behavior described above depend on fact that the feature is opt-out (is active by default). If not, the paragraph above is meaningless. This has to be yet decided/clarified. 
+
+## Communication
+
+The communication between nodes should be done over [StreamJsonRpc](https://github.com/microsoft/vs-streamjsonrpc/). The API over which two node will transfer data has to reflect inputs and outputs of RAR task as described in [docs](https://docs.microsoft.com/en-us/visualstudio/msbuild/resolveassemblyreference-task?view=vs-2019). It is possible that into the API may be added some additional data.
```

Comment:
Again, fix link

I don't think the last sentence is necessary, but if you do:
`We may augment the API with more data.` sounds better.